### PR TITLE
feat: [PLG-2329]: credit card verification flow

### PR DESCRIPTION
**Pull request by dhruvUpadhyay07**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by dhruvUpadhyay07** [comment]
retrigger cypress-cd

File: src/modules/33-auth-settings/pages/Billing/FooterCreditCard.tsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        }
+      })
+
+      //save the card details call the save api
+      await saveCard({
+        accountIdentifier: accountId,
+        creditCardIdentifier: res?.setupIntent?.payment_method as string
+      })
+
+      showSuccess('Card is validated and saved')
```
Comment (devinder-harness) [inline src/modules/33-auth-settings/pages/Billing/FooterCreditCard.tsx:70]: move to strings

**Review by devinder-harness — DISMISSED** [review]

**Review by kapil-harness — APPROVED** [review]

File: src/modules/33-auth-settings/pages/Billing/__tests__/PaymentMethods2.test.tsx
Comment (mayankVermaHarness) [inline src/modules/33-auth-settings/pages/Billing/__tests__/PaymentMethods2.test.tsx:1]: Why do we have 2 test files for PaymentMethods?

File: src/modules/33-auth-settings/strings/strings.en.yaml [hunk 2]
```diff
@@ -362,3 +361,6 @@ yesIamSure: Yes, I am very sure
 jitProvisioning: JIT Provisioning
 jitValidationKey: JIT Validation Key
 jitValidationValue: JIT Validation Value
+setAsDefaultCard: set as default card
```
Comment (mayankVermaHarness) [inline src/modules/33-auth-settings/strings/strings.en.yaml:364]:
```suggestion
setAsDefaultCard: Set as Default Card
```

**Review by devinder-harness — APPROVED** [review]

**Comment by dhruvUpadhyay07** [comment]
retrigger cypress-pipeline

**Review by kapil-harness — APPROVED** [review]

**Pull request closed without merging**